plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -340,6 +346,26 @@ public PostgreSqlClient(
                         .build());
     }
 
+    @VisibleForTesting
+    Optional<String> mapTypeSignature(ConnectorSession connectorSession, Type type)
+    {
+        if (type == TIMESTAMP_NANOS || type == TIMESTAMP_TZ_NANOS) {
+            return Optional.empty();
+        }
+
+        if (type instanceof ArrayType) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.ofNullable(this.toWriteMapping(connectorSession, type).getDataType());
```

Comment:
Delegating to `toWriteMapping` isn't generally safe correct, hence the exclusions above.
The code may be producing correct results today (i haven't checked), but is error prone: any new addition or change in `toWriteMapping` may the code to work incorrectly, without any test failing. Thus, the delegation here is error-prone.

For example, let's say i add support for TIMESTAMP_PICOS in `toWriteMapping`... (actually, it's already there...).

Instead,

1. let's call this method `mapTypeForCast`
2. let's explicitly define which types we support for CAST pushdown; for these, we don't need to delegate to `toWriteMapping`, since we know the mapping
3. let's have a test coverage for every supported type.
4. Maybe `io.trino.testing.BaseConnectorTest#testDataMappingSmokeTest` could exercise query correctness (obvious when pushdown doesn't happen, but important to test when it does)